detection_rules/rule.py

```diff
@@ -251,7 +251,7 @@ class ThresholdCardinality:
             field: str
             value: definitions.ThresholdValue
 
-        field: List[str]
+        field: List[Optional[definitions.NonEmptyStr]]
```

Comment:
don't you mean this? otherwise you could allow `[None]` which seems odd
```suggestion
        field: List[definitions.NonEmptyStr]
```